doc/man7/pbkdf2.pod

```diff
@@ -0,0 +1,129 @@
+=pod
```

Comment:
I'm fine either way -- but I'm holding off on this until you've come to consensus, if that's okay? In any case if we move the PBKDF2 docs to man3, I'd also apply the same to the scrypt man page in a different PR (that's already on master as man7).